### ISSUE TITLE
Slim down default environment spun up by "make dev" 

### ISSUE THREAD

**Issue by zenmonkeykstop** [freedomofpress/securedrop#6620]
## Description

(As suggested by @legoktm)

Slim down default environment spun up by "make dev", removing dependencies and setup required for testing. This would help to speed up dev env provisioning.

(This could be done via separate Dockerfiles, but another approach might be to have a single file with named phases, and to call docker build with --target as appropriate.)

## User Research Evidence
n/a, as suggested by @legoktm

## User Stories
As a developer, I want to use minimal time and resources when spinning up a dev environment.

**Comment by legoktm** [comment]
We could also have CI publish the dev container to e.g. quay.io so people only need to download it on first run instead of "building" it.

My PR so far:
```
localhost/securedrop-full-focal-py3                       latest               ce9e5513fa26  About a minute ago  2.9 GB
localhost/securedrop-test-focal-py3                       latest               788b40289994  31 minutes ago      1.17 GB
```

**Comment by zenmonkeykstop** [comment]
Closed by #6625 - I don't think there's a need to get into the publishing to quay.io bit for this round.
